packages/ui-components/src/components/TextInput/TextInput.test.tsx

```diff
@@ -0,0 +1,186 @@
+    expect(screen.getByRole("spinbutton")).toHaveAttribute("type", "number")
+  })
+
+  test("renders a disabled input as passed", () => {
+    render(<TextInput disabled />)
+    expect(screen.getByRole("textbox")).toBeInTheDocument()
+    expect(screen.getByRole("textbox")).toBeDisabled()
+  })
+
+  test("renders a readonly input as passed", () => {
+    render(<TextInput readOnly />)
+    expect(screen.getByRole("textbox")).toBeInTheDocument()
+    expect(screen.getByRole("textbox")).toHaveAttribute("readonly")
+  })
+
+  test("renders a focussed input as passed", () => {
+    render(<TextInput autoFocus />)
+    expect(screen.getByRole("textbox")).toBeInTheDocument()
+    expect(screen.getByRole("textbox")).toHaveFocus()
+  })
+
+  test("renders an invalid input as passed", () => {
+    render(<TextInput invalid />)
+    expect(screen.getByRole("textbox")).toBeInTheDocument()
+    expect(screen.getByRole("textbox")).toHaveClass("juno-textinput-invalid")
+    expect(screen.getByTitle("Dangerous")).toBeInTheDocument()
+  })
+
+  test("renders a valid input as passed", () => {
+    render(<TextInput valid />)
+    expect(screen.getByRole("textbox")).toBeInTheDocument()
+    expect(screen.getByRole("textbox")).toHaveClass("juno-textinput-valid")
+    expect(screen.getByTitle("CheckCircle")).toBeInTheDocument()
+  })
+
+  test("renders a helptext as passed", () => {
+    render(<TextInput helptext="this is a helptext" />)
+    expect(document.querySelector(".juno-form-hint")).toBeInTheDocument()
+    expect(document.querySelector(".juno-form-hint")).toHaveClass("juno-form-hint-help")
+    expect(document.querySelector(".juno-form-hint")).toHaveTextContent("this is a helptext")
+  })
+
+  test("renders a successtext as passed and validates the Element", () => {
+    render(<TextInput successtext="great success!" />)
+    expect(document.querySelector(".juno-form-hint")).toBeInTheDocument()
+    expect(document.querySelector(".juno-form-hint")).toHaveClass("juno-form-hint-success")
+    expect(document.querySelector(".juno-form-hint")).toHaveTextContent("great success!")
+    expect(screen.getByRole("textbox")).toHaveClass("juno-textinput-valid")
+  })
+
+  test("renders an errortext as passed and invalidates the Element", () => {
+    render(<TextInput errortext="this is an error!" />)
+    expect(document.querySelector(".juno-form-hint")).toBeInTheDocument()
+    expect(document.querySelector(".juno-form-hint")).toHaveClass("juno-form-hint-error")
+    expect(document.querySelector(".juno-form-hint")).toHaveTextContent("this is an error!")
+    expect(screen.getByRole("textbox")).toHaveClass("juno-textinput-invalid")
+  })
+
+  test("fires onChange handler as passed", () => {
+    const handleChange = vi.fn()
+    render(<TextInput onChange={handleChange} data-testid="my-input" />)
+    const textinput = screen.getByRole("textbox")
+    fireEvent.change(textinput, { target: { value: "a" } })
```

Comment:
please wrap all events that you use with act or waitFor otherwise they throw a race condition that breaks all our test case

Reply:
Done - sorry I forgot to include file in PR